[COREAPP-225] doc: add Notifications on Sourcegraph cloud (#25092)
sourcegraph/sourcegraph: doc/cloud/nofitications_on_cloud.md

# Notifications on Sourcegraph cloud

There are various kinds of notifications to indicate status of your repositories on Sourcegraph cloud:

> 1. **No repositories:** You have not added any repository, try [Adding repositories to Sourcegraph cloud](../code_search/how-to/adding_repositories_to_cloud.md).

Reviewer: ```suggestion
1. **No repositories:** You have not added any repositories. Try [Adding repositories to Sourcegraph cloud](../code_search/how-to/adding_repositories_to_cloud.md).
```

> 1. **Repositories cloning:** Some or all of your repositories are currently being cloned into Sourcegraph cloud. At this stage, search results are computed on-the-fly over your repositories which could be slow, and you may not get search results from all of your repositories.

Reviewer: ```suggestion
1. **Repositories cloning:** Some or all of the repositories you've added to Sourcegraph are currently being cloned. At this stage, search results are computed on-the-fly over your repositories. Note that this may be slow, and you may not get search results across all of the repositories you've added to Sourcegraph.
```

> 1. **Repositories indexing:** Some or all of your repositories are currently being indexed. At this stage, search results are coming back much faster for your repositories.

Reviewer: ```suggestion
1. **Repositories indexing:** Some or all of your repositories are currently being indexed. Search results will return quickly across all of your repositories.
```

Reviewer 2: Should we instead say, that `Search results will return quickly across all of your indexed repositories.`?
Because for those repositories that are still in the indexing queue, this is not really true.

Reviewer: Ooh, great call!

Reviewer: Updated original to get this idea across: searches will target all of the repositories, even unindexed ones, but they'll be faster when indexed.

> 1. **Everthing goes well:** All of your repositories are successfully cloned and indexed, search results are complete and coming back very fast.

Reviewer 2: Do we actually have a notification that says `Everything goes well`? Also, there seems to be a typo:
```suggestion
1. **Everything goes well:** All of your repositories are successfully cloned and indexed, search results are complete and coming back very fast.
```

> 1. **Everything goes well:** All of your repositories are successfully cloned and indexed, search results are complete and coming back very fast.

Reviewer: ```suggestion
1. **Everything is good:** All of the repositories you've added to Sourcegraph have been cloned and indexed. Search results are fast across all of your repositories.
```

> 1. **Everything is good:** All of the repositories you've added to Sourcegraph have been cloned and indexed. Search results are fast across all of your repositories.

Reviewer: ```suggestion
1. **Everything is good:** All of the repositories you've added to Sourcegraph have been cloned and indexed. Search results will be fast across all of your repositories.
```

> 1. **Something went wrong:** Something unexpected happened during repositories cloning stage, please refer to the [Troubleshooting](#Troubleshooting) section for possible solutions.

Reviewer: ```suggestion
1. **Something went wrong:** Something unexpected happened while cloning the repositories you added to Sourcegraph. Please refer to the [Troubleshooting](#Troubleshooting) section for possible solutions.
```

### Troubleshooting

> 1. **Sourcegraph.com OAuth application is revoked on the code host:** You may have accidentally revoked the Sourcegraph.com OAuth application on the code host, if you do not see the Sourcegraph.com OAuth application appear in your authorized OAuth applications list, try repeating [Adding repositories to Sourcegraph cloud](../code_search/how-to/adding_repositories_to_cloud.md).

Reviewer: ```suggestion
1. **Sourcegraph.com OAuth application is revoked on the code host:** Check whether the Sourcegraph.com OAuth application on the code host appears in your authorized OAuth applications list. If it does not appear, it may have been revoked. Try repeating [Adding repositories to Sourcegraph cloud](../code_search/how-to/adding_repositories_to_cloud.md).
```

Reviewer 2: Changed the text with the same context, but slightly different wording.

> 1. **API rate limit exceeded:** The API rate limit quota of GitHub.com [shares among all OAuth applications authorized by that user, personal access tokens owned by that user, and requests authenticated with that user's username and password](https://docs.github.com/en/developers/apps/building-github-apps/rate-limits-for-github-apps#normal-user-to-server-rate-limits). Sourcegraph cloud consumes very low usage of API rate limit quota, this may be expected when you're having heavy workloads for the hour, and should be resolved automatically once the API rate limit quota is refilled in the next hour for GitHub.com.

Reviewer: ```suggestion
1. **API rate limit exceeded:** GitHub.com's API rate limit quota is [shared among all OAuth applications authorized by a user, personal access tokens owned by that user, and requests authenticated with that user's username and password](https://docs.github.com/en/developers/apps/building-github-apps/rate-limits-for-github-apps#normal-user-to-server-rate-limits). While Sourcegraph cloud consumes very low usage of API rate limit quota, if you have an unusually heavy workload during a given period of time, this rate limit may be exceeded. If this happens, the error should be resolved automatically within the next hour once the GitHub.com API rate limit quota is renewed.
```

Reviewer 2: Changed the text with the same context, but slightly different wording. It reads better to me this way.
